src/Fields/interpolate.jl

```diff
@@ -1,16 +1,16 @@
 using Oceananigans.Grids: XRegRectilinearGrid, YRegRectilinearGrid, ZRegRectilinearGrid
+using CUDA: allowscalar 
 
 @inline function linear_interpolate_sorted_vector(vec, val)
 
-    @CUDA.allowscalar begin
-        y₂ = searchsortedfirst(vec, val)
-        y₁ = searchsortedlast(vec,  val)
+    allowscalar(true)
+    y2 = searchsortedfirst(vec, val)
+    y1 = searchsortedlast(vec,  val)
+    x2 = vec[y2]
+    x1 = vec[y1]
+    allowscalar(false)
```

Comment:
🤦‍♂️

Reply:
Is this insanely slow on the GPU?

Reply:
for sure it is, not in case of `LagrangianParticles` because the `interpolate` function is called from a kernel